client/state/comments/selectors/get-comment-by-id.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { filter, find, flatMap } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { deconstructStateKey, getErrorKey } from 'state/comments/utils';
+
+import 'state/comments/init';
+
+export function getCommentById( { state, commentId, siteId } ) {
+	const errorKey = getErrorKey( siteId, commentId );
+	if ( ( state.comments.errors ?? {} )[ errorKey ] ) {
+		return state.comments.errors[ errorKey ];
+	}
+
+	const commentsForSite = flatMap(
+		filter( state.comments.items ?? [], ( comment, key ) => {
```

Comment:
I wonder if `state.comments.items` can ever be nullish, given that its default state is `{}`. Also, a side note, it feels odd that the items reducer is structured as an object but we default to an array here. Do you know if there's a reason for that?

Reply:
It doesn't look like it's possible for it to ever become nullish. Good catch! I'm not sure why we were defaulting to an array instead of an object; my guess is that the structure evolved over time, and the selector wasn't updated accordingly. And I just translated the existing code instead of looking deeper like you did 😄